### PR TITLE
MANTA-4978 Settings of buckets components and storinfo should be adjusted based on deployment size MANTA-5234 Remove obsolete storage SAPI metadata from service config files

### PR DESCRIPTION
**Pull request by askfongjojo**
Tested changes in coal with manta-deployment zone image that contains this PR, ran `manta-init` for all 3 sizes (coal/lab/production) and verified services SAPI params/metadata.

### CODE REVIEW

File: config/services/buckets-mdapi/service.json [hunk 1]
```diff
@@ -4,6 +4,6 @@
 		"ram": 256
 	},
 	"metadata": {
-		"MORAY_MAX_PG_CONNS": 16
+		"BUCKETS_MDAPI_CUEBALL_MAX_CONNECTIONS": 16
```
Comment (trentm) [inline config/services/buckets-mdapi/service.json:7]: You want to reduce the cueball conns to 16 for *coal*?
I think this'll have the effect of setting it to 16 for *production* size as well, down from the hardcoded default of 64 in the sapi template:

```
[cueball]
{{#BUCKETS_MDAPI_CUEBALL_MAX_CONNECTIONS}}
max_connections = {{BUCKETS_MDAPI_CUEBALL_MAX_CONNECTIONS}}
{{/BUCKETS_MDAPI_CUEBALL_MAX_CONNECTIONS}}
{{^BUCKETS_MDAPI_CUEBALL_MAX_CONNECTIONS}}
max_connections = 64
{{/BUCKETS_MDAPI_CUEBALL_MAX_CONNECTIONS}}
```

File: config/services/storinfo/service.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"params": {
 		"networks": [ "manta", "admin" ],
-		"ram": 256
+		"ram": 1024
```
Comment (trentm) [inline config/services/storinfo/service.json:4]: FWIW in my COAL (granted idle), my storinfo is happily at only 90MB of RAM:

```
[root@headnode (coal) ~]# zonememstat -a
                                 ZONE                                             ALIAS  RSS(MB) CAP(MB)    NOVER  POUT(MB) SWAP%
...
 31dea065-d727-475b-80aa-856d4630e431            storinfo.coalregion.joyent.us-31dea065       90    256        0         0 24.353
```

Do we know that it needs to be bigger for *COAL*?  I guess @khangngu had some experience... at "lab" size?
If it is a pain to come up with a number now, we can always come back and trim down *coal* sizes separately.

File: config/services/storage/service.json [hunk 3]
```diff
@@ -4,8 +4,5 @@
 		"ram": 256,
```
Comment (trentm) [inline config/services/storage/service.json:4]: I wonder if *this* could use a bump. See this (NOVERs for a storage zone) in nightly-2, for example:

```
[root@headnode (nightly-2) ~]# zonememstat -a
                                 ZONE                                         ALIAS  RSS(MB) CAP(MB)    NOVER  POUT(MB) SWAP%
...
 0a1dc7ec-5c44-420e-a22b-a4c7dc5946b2            storage.nightly.joyent.us-0a1dc7ec      236    256      151        26 44.3413
 9707c6ef-1c8a-4f6b-8861-eba8bc3cb321           storinfo.nightly.joyent.us-9707c6ef      178    256        0         0 29.7279
```

Comment (askfongjojo) [reply config/services/storage/service.json]: In perf lab, the rss is somewhere between 250-300. I've bumped it up to 512 which should be suffix for most dev/test labs.

File: config/services/buckets-mdapi/service.json (hunk 1 above)
Comment (askfongjojo) [reply config/services/buckets-mdapi/service.json]: My bad, yes it does change max connections to 16 as well for prod. Checking the config in the ticket again, we can just make it 32 for coal and keep it at 64 for other sizes.

File: config/services/storinfo/service.json (hunk 2 above)
Comment (askfongjojo) [reply config/services/storinfo/service.json]: Checking the labs which have up to 6 storage zones, storinfo still stay below 256. I've walked back on the change for coal (and lab).

File: config/services/buckets-mdapi/service.json.lab [hunk 4]
```diff
@@ -1,5 +1,8 @@
 {
 	"params": {
 		"ram": 1024
+        },
+        "metadata": {
+                "BUCKETS_MDAPI_CUEBALL_MAX_CONNECTIONS": 64
```
Comment (trentm) [inline config/services/buckets-mdapi/service.json.lab:6]: nit: spaces in a file that was using tabs. I'm fine one way or the other, but mixed tabs and spaces is pretty messy.

File: config/services/buckets-mdapi/service.json.production [hunk 5]
```diff
@@ -2,5 +2,8 @@
 	"params": {
 		"ram": 32768,
                 "quota": 100
+        },
+        "metadata": {
+                "BUCKETS_MDAPI_CUEBALL_MAX_CONNECTIONS": 64
```
Comment (trentm) [inline config/services/buckets-mdapi/service.json.production:7]: nit: spaces vs tabs here too

File: config/services/storinfo/service.json.production [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+	"params": {
+		"ram": 16384,
```
Comment (trentm) [inline config/services/storinfo/service.json.production:3]: nit: mixed tabs and spaces in this file

Comment (askfongjojo) [reply config/services/storinfo/service.json.production:3]: fixed

File: config/services/buckets-mdapi/service.json.lab (hunk 4 above)
Comment (askfongjojo) [reply config/services/buckets-mdapi/service.json.lab]: fixed

File: config/services/buckets-mdapi/service.json.production (hunk 5 above)
Comment (askfongjojo) [reply config/services/buckets-mdapi/service.json.production]: fixed

File: config/services/buckets-mdapi/service.json.lab [hunk 7]
```diff
@@ -1,5 +1,8 @@
 {
 	"params": {
 		"ram": 1024
+	},
+	"metadata": {
+		"BUCKETS_MDAPI_CUEBALL_MAX_CONNECTIONS": 64
```
Comment (trentm) [inline config/services/buckets-mdapi/service.json.lab:6]: Khang *had* suggested 48 here in the description for MANTA-4978, and then you said "16" for lab in a comment:

> For buckets-mdapi, we can make the number of connections similar to those of moray, i.e. 16 for coal/lab and 64 (default) for production.

Comment (askfongjojo) [reply config/services/buckets-mdapi/service.json.lab:6]: The latest is now 32 for coal, 64 for lab, and 64 for production. Looking at the number of mdapi instances we'd normally provisioned per shard and the postgresql settings he referenced for different sizes:

postgresql.manta.coal.conf:max_connections = 100 (1 mdapi per shard)
postgresql.manta.lab.conf:max_connections = 210 (1-2 mdapi per shard)
postgresql.manta.production.conf:max_connections = 1000 (3-6 mdapi per shard)

we can actually keep the "lab" one to 64.

**Pull request merged**